epictrack-web/src/components/shared/controlledInputComponents/ControlledTextField.tsx

```diff
@@ -31,6 +33,9 @@ const ControlledTextField: FC<IFormInputProps> = ({
           {...field}
           {...otherProps}
           onChange={(e) => {
+            if (maxLength && e.target.value.length > maxLength) {
+              return;

```

Comment:
I think you don't want to check the max length like this. You should be able to set it like this inside the common control
inputProps={{
            maxLength: 12,
          }}